package.json

```diff
@@ -1,34 +1,48 @@
 {
   "private": true,
   "scripts": {
-    "programs:build": "zx ./scripts/program/build.mjs",
-    "programs:test": "zx ./scripts/program/test.mjs",
-    "programs:clean": "zx ./scripts/program/clean.mjs",
-    "programs:format": "zx ./scripts/program/format.mjs",
-    "programs:lint": "zx ./scripts/program/lint.mjs",
-    "solana:check": "zx ./scripts/check-solana-version.mjs",
-    "solana:link": "zx ./scripts/link-solana-version.mjs",
+    "solana:check": "tsx ./scripts/helpers/check-solana-version.mts",
+    "solana:link": "tsx ./scripts/helpers/link-solana-version.mts",
     "generate": "pnpm generate:clients",
-    "generate:clients": "zx ./scripts/generate-clients.mjs",
-    "validator:start": "zx ./scripts/start-validator.mjs",
+    "generate:clients": "tsx ./scripts/helpers/generate-clients.mts",
+    "validator:start": "tsx ./scripts/helpers/start-validator.mts",
     "validator:restart": "pnpm validator:start --restart",
-    "validator:stop": "zx ./scripts/stop-validator.mjs",
-    "clients:js:format": "zx ./scripts/client/format-js.mjs",
-    "clients:js:lint": "zx ./scripts/client/lint-js.mjs",
-    "clients:js:publish": "zx ./scripts/client/publish-js.mjs",
-    "clients:js:test": "zx ./scripts/client/test-js.mjs",
-    "clients:rust:format": "zx ./scripts/client/format-rust.mjs",
-    "clients:rust:lint": "zx ./scripts/client/lint-rust.mjs",
-    "clients:rust:publish": "zx ./scripts/client/publish-rust.mjs",
-    "clients:rust:test": "zx ./scripts/client/test-rust.mjs"
+    "validator:stop": "tsx ./scripts/helpers/stop-validator.mts",
+    "program:build": "tsx ./scripts/rust.mts build-sbf program --features bpf-entrypoint",
+    "program:format": "tsx ./scripts/rust.mts format program",
+    "program:lint": "tsx ./scripts/rust.mts lint program",
+    "program:lint:clippy": "tsx ./scripts/rust.mts lint-clippy program",
+    "program:lint:docs": "tsx ./scripts/rust.mts lint-docs program",
+    "program:lint:features": "tsx ./scripts/rust.mts lint-features program",
+    "program:test": "tsx ./scripts/rust.mts test program",
+    "js:format": "tsx ./scripts/js.mts format clients/js",
+    "js:lint": "tsx ./scripts/js.mts lint clients/js",
+    "js:publish": "tsx ./scripts/js.mts publish clients/js",
+    "js:test": "tsx ./scripts/js.mts test clients/js",
+    "rust:format": "tsx ./scripts/rust.mts format clients/rust",
+    "rust:lint": "tsx ./scripts/rust.mts lint clients/rust",
+    "rust:lint:clippy": "tsx ./scripts/rust.mts lint-clippy clients/rust",
+    "rust:lint:docs": "tsx ./scripts/rust.mts lint-docs clients/rust",
+    "rust:lint:features": "tsx ./scripts/rust.mts lint-features clients/rust",
+    "rust:publish": "tsx ./scripts/rust.mts publish clients/rust",
+    "rust:test": "tsx ./scripts/rust.mts test clients/rust",
```

Comment:
Is there a reason that these aren't prefixed with `clients`? Ie. `clients:js:format`? The reason I ask is that it'll be inconsistent with the other repos

Reply:
I think this was the new "template" that we agreed, dropping the "client". Only the system repository is following this at the moment, so we could revert this pattern.

Reply:
Hmm yeah I think this is something that wasn't obvious with the `system` repo because the program doesn't live in the repo but I agree with Jon, this lack consistency and isn't predictable.

I really like our latest refactoring where, for instance, everything Rust-related is in its own script and then you have sub commands within that script.

Maybe we should lean into that more in the pnpm scripts as well. We could have a simple `rust` script and let the user of that script to provide the arguments.

So instead of something like `pnpm program:format` we would do `pnpm rust format program`. That way, there are less "indirections" between the scripts and how they are invoked. What do you think?

Reply:
The simplicity of running the commands is definitely nice, but on the flip-side you lose the "self-documenting" nature of package.json, which currently tells you everything that you should be able to run successfully in a repo.

I don't have a very strong preference, but I would stick with the `pnpm program:format` solely for self-documentation.